stra.c

```diff
@@ -12,3 +12,45 @@ CHAR8 *strstra(const CHAR8 *input, const CHAR8 *find) {
 	} while (*(input++) != '\0');
 	return NULL;
 }
+
+
+/* UEFI shell populates LoadOptions and LoadOptionsSize in some odd ways.
+ * One of which is that argv[0] (the name of the efi being executed) is
+ * present in the LoadOptions.
+ *
+ * This function simply removes the first token of the command line
+ * if that token contains ends with '.efi' or '.EFI'.
+ *
+ * https://github.com/puzzleos/stubby/issues/19
+ */
```

Comment:
I wonder if there is a more reliable way to detect this.  Does UEFI tell us who called us, so we could detect that the shell was our caller?

Reply:
well, i admit to taking a shortcut.
There is a longer / more correct path in #18, which basically copies what shim does.
See comments at https://github.com/rhboot/shim/blob/4fd484e4c29364b4fdf4d043556fa0a210c5fdfc/load-options.c#L306 

#18 has the pain of different licenses on that code, so i just took the cheap way out.

It does admittedly fail if you did:

    kernel.efi something.efi console=ttyS0

where `something.efi` was actually intended to be passed to the kernel. (but in this case, as it is right now stubby would have rejected that anyway because `something.efi` isn't allowed).

Reply:
I'm ok with the shortcut, but let's leave a note or an issue to look for another way, maybe?

The shell doesn't require the filenames it executes to end with efi, right?  So another way this could fail is if someone renames kernel.efi to just runme, and then runs that?  It's mostly a "doctor it hurts when I do this", but ...